src/goto-symex/symex_main.cpp

```diff
@@ -519,3 +525,115 @@ void goto_symext::symex_step(
     UNREACHABLE;
   }
 }
+
+/// Collect all sub-expressions of an expression that are pointer-typed symbols
+/// \param expr: The expression to examine
```

Comment:
Document what state `expr` might be in -- the one from the GOTO path is a plain GOTO-expr (i.e. not clean_expr'd, not renamed), the one from the ASSUME path is both cleaned and renamed to L2.

Reply:
I've rebased on top of  #4309 and decided to use the condition before it is cleaned and renamed.